src/file_operations.js

```diff
@@ -60,22 +58,28 @@ const scanCourses = async (inputPath, outputPath, options = {}) => {
           .map(path => directoryExists(path))
       )
     ).filter(Boolean).length
-  const coursesPath = path.join(outputPath, "courses")
-  if (numCourses > 0) {
-    // populate the course uid mapping
-    const courseUidsLookup = {}
-    for (const course of courseList) {
-      const courseUid = await getCourseUid(inputPath, course)
-      courseUidsLookup[courseUid] = course
-    }
-    console.log(`Converting ${numCourses} courses to Hugo markdown...`)
-    progressBar.start(numCourses, 0)
-    for (const course of courseList) {
-      await scanCourse(inputPath, coursesPath, course, courseUidsLookup)
-      progressBar.increment()
-    }
-  } else {
+  if (numCourses === 0) {
     console.log(NO_COURSES_FOUND_MESSAGE)
+    return
+  }
+
+  // populate the course uid mapping
+  const courseUidsLookup = {}
+  for (const course of courseList) {
+    if (!(await directoryExists(path.join(inputPath, course)))) {
```

Comment:
When I ran this against my `aws sync`'d local copy of `open-learning-course-data`, I got this error:

```
gumaerc@gumaerc-laptop:~/Code/ocw-to-hugo$ node . -i ../hugo-course-publisher/private/courses -o ../hugo-course-publisher-2/site/content
Error: Error: Missing course directory for edx_courses.json
    at scanCourses (/home/gumaerc/Code/ocw-to-hugo/src/file_operations.js:70:13)
    at async run (/home/gumaerc/Code/ocw-to-hugo/src/bin/index.js:66:3)
```
It seems that `directoryExists` is returning true even if it's a file and not a directory.  While iterating `courseList`, we should skip entries that are not directories.

Reply:
I should note though that it ran fine after removing the stray json files from that directory, although it should ignore them.  Also specifying courses that didn't exist in a `courses.json` file errored as expected.

Reply:
Good catch, should be fixed now. I added a file to `test_data/courses` to make sure the tests detect this issue in the future